src/EA.Iws.Web.Tests.Unit/ViewModels/AdminExportNotificationMovements/CreateViewModelTests.cs

```diff
@@ -0,0 +1,229 @@
+        }
+
+        [Fact]
+        public void WasteReceivedDateCanBeInThePast()
+        {
+            var model = CreateTestModel(30, 5, 2016, "RECEIVED_PAST");
+            Assert.Empty(ViewModelValidator.ValidateViewModel(model));
+        }
+
+        [Fact]
+        public void WasteReceivedDateCanBeToday()
+        {
+            var model = CreateTestModel(1, 6, 2016, "RECEIVED");
+            Assert.Empty(ViewModelValidator.ValidateViewModel(model));
+        }
+
+        [Fact]
+        public void WasteReceivedDateNotBeforeActualShipmentDate()
+        {
+            var model = CreateTestModel(31, 5, 2016, "RECEIVED");
+            Assert.NotEmpty(ViewModelValidator.ValidateViewModel(model));
+        }
+
+        [Fact]
+        public void WasteReceivedDateSameasActualShipmentDate()
+        {
+            var model = CreateTestModel(1, 6, 2016, "RECEIVED");
+            Assert.Empty(ViewModelValidator.ValidateViewModel(model));
+        }
+
+        [Fact]
+        public void WasteReceivedDateCannotBeInTheFuture()
+        {
+            var model = CreateTestModel(30, 7, 2016, "RECEIVED");
+            Assert.NotEmpty(ViewModelValidator.ValidateViewModel(model));
+        }
+
+        [Fact]
+        public void WasteRecoveredDateCanBeInThePast()
+        {
+            var model = CreateTestModel(30, 5, 2016, "RECOVERED_PAST");
+            Assert.Empty(ViewModelValidator.ValidateViewModel(model));
+        }
+
+        [Fact]
+        public void WasteRecoveredDateCanBeToday()
+        {
+            var model = CreateTestModel(1, 6, 2016, "RECOVERED");
+            Assert.Empty(ViewModelValidator.ValidateViewModel(model));
+        }
+
+        [Fact]
+        public void WasteRecoveredDateNotBeforeWasteReceivedDate()
+        {
+            var model = CreateTestModel(31, 5, 2016, "RECOVERED");
+            Assert.NotEmpty(ViewModelValidator.ValidateViewModel(model));
+        }
+
+        [Fact]
+        public void WasteRecoveredDateSameasWasteReceivedDate()
+        {
+            var model = CreateTestModel(1, 6, 2016, "RECOVERED");
+            Assert.Empty(ViewModelValidator.ValidateViewModel(model));
+        }
+
+        [Fact]
+        public void WasteRecoveredDateCannotBeInTheFuture()
+        {
+            var model = CreateTestModel(30, 7, 2016, "RECOVERED");
+            Assert.NotEmpty(ViewModelValidator.ValidateViewModel(model));
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="day"></param>
+        /// <param name="month"></param>
+        /// <param name="year"></param>
+        /// <param name="type">
+        /// PRE_NOTIFICATION - Prenotification Date
+        /// A - Actual shipment Date
+        /// RECEIVED - Waste received Date
+        /// RECOVERED - Waste recovered Date
+        /// </param>
+        /// <returns></returns>
+        internal CreateViewModel CreateTestModel(int day, int month, int year, string type)
```

Comment:
I think it would be better to break down this method into smaller ones, e.g. CreateViewModelForPrenotificationDate, CreateViewModelForActualDate, etc. It's quite long and relies on a magic string for the setup which should be avoided if possible.